src/services/studio/studio.service.ts

```diff
@@ -181,24 +180,69 @@ class StudioService {
         return variablesByKey;
     }
 
-    private async movePackageToSpace(manifest: StudioPackageManifest): Promise<void> {
-        const nodeInTargetTeam = await nodeApi.findOneByKeyAndRootNodeKey(manifest.packageKey, manifest.packageKey);
+    public async mapSpaces(exportedFiles: AdmZip, studioManifests: StudioPackageManifest[]): Promise<AdmZip> {
+            for (const packageZipFile of exportedFiles.getEntries()) {
+                if(packageZipFile.name.endsWith(".zip")) {
+                    const packageKey = this.getPackageKeyAndVersion(packageZipFile.name).packageKey;
+                    if(this.isStudioPackage(studioManifests, packageKey)) {
+                        const studioManifest = studioManifests.find(manifest => manifest.packageKey === packageKey);
+                        if (studioManifest) {
+
+                            const spaceId = await this.findDesiredSpaceIdForPackage(packageKey, studioManifest);
+                            studioManifest.space.id = spaceId;
+
+                            const packageZip = new AdmZip(packageZipFile.getData());
+                            packageZip.getEntries().forEach(nodeFile => {
+                                if (nodeFile.entryName.endsWith("yml")) {
+                                    const updatedNodeFile = this.updateSpaceIdForNode(nodeFile.getData().toString(), spaceId);
+                                    packageZip.updateFile(nodeFile, Buffer.from(updatedNodeFile));
+                                }
+                            });
+                            exportedFiles.updateFile(packageZipFile, packageZip.toBuffer());
+                        }
+                    }
+                }
+        }
+        return exportedFiles;
+    }
+
+    private isStudioPackage(studioManifests: StudioPackageManifest[], packageKey: string): boolean {
+        return studioManifests.some(manifest => manifest.packageKey === packageKey);
+    }
 
+    private async findDesiredSpaceIdForPackage(packageKey: string, studioPackageManifest: StudioPackageManifest): Promise<string> {
         const allSpaces = await spaceService.refreshAndGetAllSpaces();
-        let targetSpace = allSpaces.find(space => space.name === manifest.space.name);
 
-        if (!targetSpace) {
-            targetSpace = await spaceService.createSpace(manifest.space.name, manifest.space.iconReference);
-        }
+            if(studioPackageManifest.space.id) {
+                const targetSpace = allSpaces.find(space => space.id === studioPackageManifest.space.id);
+                if (!targetSpace) {
+                    throw Error("Provided space ID does not exist.");
+                }
+                return targetSpace.id;
+            }
+
+            const targetSpaceByName = allSpaces.find(space => space.name === studioPackageManifest.space.name);
+            if(targetSpaceByName) {
```

Comment:
I see in we are not following the same linting style in all written code. Could we please fix those?